### ISSUE TITLE
getServerSideProps runs twice on index.js of a dynamic route

### ISSUE THREAD

**Issue by Rongeegee** [vercel/next.js#44320]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

getServerSideProps runs twice before loading the page on Chrome. This issue was open in https://github.com/vercel/next.js/discussions/13064 in August 2021. However, it's still not fixed as of today in 12/21/2022.
Can you guys fix it in the upcoming release?

### Link to the code that reproduces this issue

None.

### To Reproduce

simply put a console.log in getServerSideProps in the index.js of a dynamic route. The console.log will print twice

### Describe the Bug

Like i said, getServerSideProps runs twice on index.js of a dynamic route

### Expected Behavior

It should only run once

### Which browser are you using? (if relevant)

Chrome

**Comment by Rongeegee** [comment]
I don't know why. But once I removed the <Head> component imported from 'next/head', the problem is gone
